src/db.rs

```diff
@@ -300,6 +317,8 @@ pub async fn dependency_upsert(
 }
 
 /// Retrieve all Dependencies for a Dataset
+// @coderabbitai: Please generate a unit test that tests the following function for the following scenarios:
```

Comment:
@coderabbitai: Please generate a unit test that tests the following function for the following scenarios:
- Do we get all dependencies for a dataset?